contrib/generate-struct.py

```diff
@@ -0,0 +1,385 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name,missing-docstring
+#
+# Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import sys
+import argparse
+
+from enum import StrEnum
+from typing import Optional, List, Tuple
+
+
+class Endian(StrEnum):
+    NATIVE = "native"
+    LITTLE = "le"
+    BIG = "be"
+
+
+class Type(StrEnum):
+    U8 = "u8"
+    U16 = "u16"
+    U24 = "u24"
+    U32 = "u32"
+    U64 = "u64"
+    STRING = "s"
+    GUID = "guid"
+
+
+class Item:
+    def __init__(self):
+        self.element_id: str = ""
+        self.type: Type = Type.U8
+        self.default: Optional[str] = None
+        self.constant: Optional[str] = None
+        self.endian: Endian = Endian.NATIVE
+        self.multiplier: int = 0
+        self.offset: int = 0
+
+    @property
+    def size(self) -> int:
+        multiplier = self.multiplier
+        if not multiplier:
+            multiplier = 1
+        if self.type in [Type.U8, Type.STRING]:
+            return multiplier
+        if self.type == Type.U16:
+            return multiplier * 2
+        if self.type == Type.U24:
+            return multiplier * 3
+        if self.type == Type.U32:
+            return multiplier * 4
+        if self.type == Type.U64:
+            return multiplier * 8
+        return 0
+
+    @property
+    def endian_glib(self) -> str:
+        if self.endian == Endian.LITTLE:
+            return "G_LITTLE_ENDIAN"
+        if self.endian == Endian.BIG:
+            return "G_BIG_ENDIAN"
+        return "G_BYTE_ORDER"
+
+    @property
+    def type_glib_mut(self) -> str:
+        tmp_type = ""
+        if self.type == Type.U8:
+            tmp_type = "guint8"
+        elif self.type == Type.STRING:
+            tmp_type = "gchar"
```

Comment:
just as a nit I'd say swap the order of these two so all the integers are together

Reply:
Fixed.